discovery-infra/start_discovery.py

```diff
@@ -229,12 +230,25 @@ def set_cluster_vips(client, cluster_id, machine_net):
     client.update_cluster(cluster_id, cluster_info)
 
 
-def set_cluster_machine_cidr(client, cluster_id, machine_net):
+def set_cluster_machine_cidr(client, cluster_id, machine_net, set_vip_dhcp_allocation=True):
     cluster_info = client.cluster_get(cluster_id)
-    cluster_info.vip_dhcp_allocation = True
-    cluster_info.machine_network_cidr = machine_net.cidr_v4
+    if set_vip_dhcp_allocation:
+        cluster_info.vip_dhcp_allocation = True
+    cluster_info.machine_network_cidr = machine_net.cidr_v6 if machine_net.has_ip_v6 and not machine_net.has_ip_v4 else machine_net.cidr_v4
     client.update_cluster(cluster_id, cluster_info)
 
+def _get_ip_for_single_node_v6(client, cluster_id):
```

Comment:
can we use this method ? 
https://github.com/openshift/assisted-test-infra/blob/master/discovery-infra/test_infra/helper_classes/cluster.py#L660

Reply:
i want to have only one place that get host ips

Reply:
This one gets IP that belongs to the machine CIDR

Reply:
but most of the logic is the same, so lets use the existing one and just verify machine-cidr ?

Reply:
And also I want to verify that is is IPv6.  Anyway, I will change

Reply:
This approach will prefer IPv4 over IPv6.  Not sure I want that